### ISSUE TITLE
Debug version runs, but Release crashes.

### ISSUE THREAD

**Issue by rwebber** [bakercp/ofxSMTP#17]
I can send email (with attachements) as I need if I build in debug, but if I build in Release it crashes.

 a ntdll.dll exception and its over.

The issue occurs after:
smtp.send(message);
building up the 'message' seems fine.
but I don't get to onSMTPDelivery() or onSMTPException()

for now I am continuing to build using the debug version... but any insight would be greatly appreciated.

**Comment by bakercp** [comment]
What versions of the addon, openFrameworks, OS, etc are you using?


On Tue, Apr 11, 2017, 11:00 AM Ryan Webber <notifications@github.com> wrote:

I can send email (with attachements) as I need if I build in debug, but if
I build in Release I get the following error.
Microsoft c++ exception boost::filesystem_error at memory location
0x0C.......
then a ntdll.dll exception and its over.

The issue occurs after:
smtp.send(message);
building up the 'message' seems fine.
but I don't get to onSMTPDelivery() or onSMTPException()

for now I am continuing to build using the debug version... but any insight
would be greatly appreciated.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/bakercp/ofxSMTP/issues/17>, or mute the thread

**Comment by rwebber** [comment]
OF 9.3
and the plugin (downloaded zip) I started working with it a couple weeks ago.
I don't see a version number inside. (.gitignore is date 2016-05-04, as are many files)

**Comment by bakercp** [comment]
If you are using 093 make sure you are using the stable breach of this addon.  The master branch will probably give you errors with older versions of oF.

**Comment by rwebber** [comment]
OK, I have done the following.

- grabbed the latest stable release (just in case thats not what I had)
- went back to the example-advanced that comes with it.

note: The boost error is due to a ofpath call in a GIF addon.. so I wanted to get to basics.. hence starting again with the included example.

using the example-advanced
- I can run in debug, and send simple [spacebar] and advanced [a]
-- this is true using both the external xml settings and the simple gmail settings
- if I run the release version, both simple and/or advance crash
-- the error I get is: 
Exception thrown at 0x77469EF3 (ntdll.dll) in example-advanced.exe: 0xC0000005: Access violation reading location 0x002B003F.

I'm stumped. I just dont know how to debug the release build.
Using ofLog I can output before and just after smtp.send()
Thats the best I got.

**Comment by rwebber** [comment]
I should mention that I am running:
Windows 10
Visual Studio 2015
OF 0.9.3

**Comment by bakercp** [comment]
Just for testing's sake, can you try it with the master branches of ofxSMTP and ofxSSLManager and the nightly openFrameworks? In the past when I've had ntdll problems, it was often a Poco-related bug that was fixed by updating Poco or openFrameworks.  Unfortunately I don't have access to a windows machine to test at the moment.

**Comment by rwebber** [comment]
Just tested with:
of 0.9.8 release, and the both stable releases of your plugins.
same result. Debug runs, release crashes.

I tried adding the master branches of your plugins, but ofxPoco is missing, so I assume these target the latest OF build.

Is there a version I am missing for tests?

**Comment by bakercp** [comment]
Go down to the latest nightlight builds here and try the master branch
examples from there:

http://openframeworks.cc/download/


---
http://christopherbaker.net

**Comment by rwebber** [comment]
OK, so after some more digging, I believe I have found the cause of the issue.
I have been using the very nice c++ package manager, vcpkg.
This has been a great time saver for other projects, but since I activated its global integration with Visual studio 2015, it has been adding links for a number of libraries automatically to any new projects, including projects created with OF's project generator.
So, my OF project has been accessing other versions of Boost and Poco that I have installed for use via vcpkg.
Since this linking integration is done behind the scenes, its easy to miss, and will potentially replace any lib OF uses with the local copy you have (in my case, freeimage, poco, boost, and probably more).
Doing the test with the latest nightly helped me find the issue, since poco is moved to a addon, there were different errors that pointed me in the right direction.

Thanks for the help. 
Hopefully this information might help another. (going to add this to the OF forum)
